scheduler/benchmarks/benchmarks_test.go

```diff
@@ -0,0 +1,252 @@
+package benchmarks
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/nomad/scheduler"
+)
+
+// TestProfileSchedulerExample is an example of how to write a one-off
+// cpu profile for the Nomad scheduler. The pprof profile will be
+// written to the path in the env var NOMAD_BENCHMARK_PROFILE_PATH.
+// The starting state for your implementation will depend on the
+// following environment variables:
```

Comment:
Why does this exist when the Benchmark version below allows using Go's builtin benchmark profiling functionality? Might be worth documenting when a developer should use one or the other as a starting point.

Reply:
Oh you're right... when I originally wrote these as part of #11712 I only had roughly equivalent to `TestProfileSchedulerExample` and `BenchmarkServiceScheduler`, and because the `b.Run` for `BenchmarkServiceScheduler` has to include setup code, the profiling was all off.

But now that we'll have `BenchmarkSchedulerExample` that's totally unnecessary and we can cut this out entirely. I'll remove this and add some docstring comments explaining that to `BenchmarkSchedulerExample`.